backupccl: TestExcludeDataFromBackupAndRestore is flaky

**Describe the problem**

```
--- FAIL: TestExcludeDataFromBackupAndRestore (2.26s)
    test_log_scope.go:161: test logs captured to: /Users/pavel/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/ab3eff6b7877b2d0e35a44d562d13cda/logTestExcludeDataFromBackupAndRestore620557206
    test_log_scope.go:79: use -show-logs to present logs inline
    backup_test.go:9278: 
                Error Trace:    /private/var/tmp/_bazel_pavel/9feea0c38530dc9f20105ecc8935971a/sandbox/darwin-sandbox/533/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/backup_test.go:9278
                Error:          "[]" should have 10 item(s), but has 0
                Test:           TestExcludeDataFromBackupAndRestore
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    panic.go:522: -- test log scope end --
FAIL
I230117 12:07:18.028974 1 (gostd) testmain.go:468  [T1] 1  Test //pkg/ccl/backupccl:backupccl_test exited with error code 1
```

Example failure in CI [here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8343637).

**To Reproduce**

```bash
./dev test pkg/ccl/backupccl --filter=TestExcludeDataFromBackupAndRestore --stress
```

**Environment:**
 - CockroachDB version: `master` @ 761cf72c, also tried at an older commit 01032c2240 to make sure the failure isn't too recent
 - Both on my local MacOS, and in CI

Jira issue: CRDB-23470

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -9251,17 +9251,21 @@ func TestExcludeDataFromBackupAndRestore(t *testing.T) {
 	// Set foo to exclude_data_from_backup and back it up. The ExportRequest
 	// should be a noop and backup no data.
 	sqlDB.Exec(t, `ALTER TABLE data.foo SET (exclude_data_from_backup = true)`)
-	waitForTableSplit(t, conn, "foo", "data")
 	waitForReplicaFieldToBeSet(t, tc, conn, "foo", "data", func(r *kvserver.Replica) (bool, error) {
 		if !r.ExcludeDataFromBackup() {
-			return false, errors.New("waiting for exclude_data_from_backup to be applied")
+			return false, errors.New("waiting for the range containing table data.foo to split")
 		}
 		return true, nil
 	})
-	waitForTableSplit(t, conn, "bar", "data")
-	sqlDB.Exec(t, `BACKUP DATABASE data TO $1`, localFoo)
+	waitForReplicaFieldToBeSet(t, tc, conn, "bar", "data", func(r *kvserver.Replica) (bool, error) {
+		if r.ExcludeDataFromBackup() {
+			return false, errors.New("waiting for the range containing table data.bar to split")
+		}
+		return true, nil
+	})
+	sqlDB.Exec(t, `BACKUP DATABASE data INTO $1`, localFoo)
 
-	restoreDB.Exec(t, `RESTORE DATABASE data FROM $1`, localFoo)
+	restoreDB.Exec(t, `RESTORE DATABASE data FROM LATEST IN $1`, localFoo)
 	require.Len(t, restoreDB.QueryStr(t, `SELECT * FROM data.foo`), 0)
 	require.Len(t, restoreDB.QueryStr(t, `SELECT * FROM data.bar`), 10)
 }

